### PR TITLE
Use @rpath instead of @loader_path for dynamic libraries

### PR DESCRIPTION
**Pull request by atljeremy**
**The Problem:**
Using `@loader_path` requires all project dependencies be in the same location. Considering this isn't always the case, using a more dynamic solution that allows for searching multiple paths is recommended.

**The Solution:**
It's recommended to create "Run-Path Dependent Libraries" that use the `@rpath` directive in order to allow consumers of your library to locate their dependencies dynamically at runtime. Here's more info from Apple on this topic and why it's the recommended solution vs something like `@loader_path` - https://developer.apple.com/library/mac/documentation/DeveloperTools/Conceptual/DynamicLibraries/100-Articles/RunpathDependentLibraries.html. Here's the Man Page describing both `@loader_path` and `@rpath` in more detail - https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man1/dyld.1.html.

**Additional Benefits:**
With the use of `@rpath` you will be providing support for https://github.com/Carthage/Carthage. Not everyone likes CocoaPods and how it makes changes to your .xcodeproj and then forces you to use a .xcworkspce. Carthage is a far easier and less intrusive solution to dependency management.

**Tests:**
I did  not write any new code for this PR so no new tests were added. However, I did run the tests to verify everything is still passing with the proposed changes. All tests pass.
```
Test Suite 'All tests' passed at 2016-03-28 15:02:17.883.
	 Executed 61 tests, with 0 failures (0 unexpected) in 0.304 (0.365) second
```

Please consider this update. Thanks.

### CODE REVIEW

File: Source/GTLCore.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1473,9 +1473,10 @@
 				COMBINE_HIDPI_IMAGES = YES;
 				DYLIB_COMPATIBILITY_VERSION = 1;
 				DYLIB_CURRENT_VERSION = 1;
+				DYLIB_INSTALL_NAME_BASE = "@rpath";
```
Comment (thomasvl) [inline Source/GTLCore.xcodeproj/project.pbxproj]: Does Apple document DYLIB_INSTALL_NAME_BASE or LOCAL_LIBRARY_DIR?  I'm not seeing it in a quick search of developer.apple.com.  :(

**Comment by thomasvl** [comment]
Aside - Re Carthage - PRs welcome.  We held off a while on pods trying to figure out what could be done to support as many folks as possible, but it ended up seeming like we sorta have to just add support for each system on their own.  :(  I can't say I have any direct experience with it, so we'll need suggestions from folks.

File: Source/GTLCore.xcodeproj/project.pbxproj (hunk 1 above)
Comment (atljeremy) [inline Source/GTLCore.xcodeproj/project.pbxproj]: Not that I know of. However, I was able to find a bit info on this topic. Here (http://lists.apple.com/archives/xcode-users/2014/Jul/msg00262.html) is a mailing list exchange from 2014 in which states the following:

> The Xcode 6 framework target template now sets DYLIB_INSTALL_NAME_BASE
to "@rpath"

and also...

> The advantage of @rpath is that it works wherever the library is
placed. So even if you _do_ want to install the framework at
$(INSTALL_PATH), using @rpath should be just fine.

I also found a similar Github Issue/PR for the MailCore2 project in which they state all of the same benefits to using `@rpath` - https://github.com/MailCore/mailcore2/issues/1023

Let me know if this helps any.

**Comment by atljeremy** [comment]
@thomasvl Regarding Carthage. The nice thing about Carthage, is that all you need to do to support it is use `@rpath`, for the reasons described above, and create tags for each release of your lib. That's it. Anyone, like myself, who'd like to use this library with Carthage simply adds `github "google/google-api-objectivec-client" ~> 1.0` to their `Cartfile` and runs `$ carthage update`. This clones the projects and builds the framework so you can just drop it into your project.

I use Carthage as my dependency manager so in order to get this library to work with Carthage I had to fork it, make the updates proposed in this PR, create a tag and push it to my fork. I then added `github "atljeremy/google-api-objectivec-client" ~> 1.0` to my `Cartfile` and everything works great now. I created this PR so that I don't have to worry about maintaining my own fork. I'd love to just be able to point my Cartfile to your repo and let it pull in updates as they become available.

Thanks.

**Comment by thomasvl** [comment]
So there isn't anything extra needed to pick up the SessionFetcher as a dep?  Do you just get it via the deps directory submodule and the Xcode project here then?  What if something else you needed also depended on the GTMSessionFetcher, would that cause a conflict/duplicate symbols?

**Comment by atljeremy** [comment]
> So there isn't anything extra needed to pick up the SessionFetcher as a dip?

No, Carthage will automatically resolve submodule dependencies as well as other Carthage dependencies if there is a Cartfile in the project.

> Do you just get it via the deps directory submodule and the Xcode project here then?

Yes. Carthage will resolve and clone any submodules and make sure they are in the final built framework.

> What if something else you needed also depended on the GTMSessionFetcher, would that cause a conflict/duplicate symbols?

Being that these are self contained .framework's that are being built, no, there wouldn't be any duplicate symbols if a project had dependencies that all had the same sub dependency.

**Pull request merged**